src/lib/utils/package-manifest.ts

```diff
@@ -205,7 +253,13 @@ export function getTsEntryPointForPackage(
 ): string | undefined | typeof ignorePackage {
     let packageMain = "index.js"; // The default, per the npm docs.
     let packageTypes = null;
-    if (
+    const typedocPackageConfig = extractTypedocConfigFromPackageManifest(
+        logger,
+        packageJsonPath
+    );
+    if (typedocPackageConfig?.entryPoint) {
+        packageMain = typedocPackageConfig.entryPoint;
+    } else if (
         hasOwnProperty(packageJson, "typedocMain") &&
```

Comment:
I will add the warning. I don't think there would be case where `typedocMain` is preferred, because I'm for making `typedoc` config a structure in `package.json`.

Reply:
Added tests for this as well